openedx_tagging/core/tagging/models.py

```diff
@@ -456,3 +463,103 @@ def resync(self) -> bool:
             changed = True
 
         return changed
+
+
+class OpenObjectTag(ObjectTag):
+    """
+    Free-text object tag.
+
+    Only needs a free-text taxonomy and a value to be valid.
+    """
+
+    class Meta:
+        proxy = True
+
+    @classmethod
+    def valid_for(cls, taxonomy: Taxonomy = None, **kwargs) -> bool:
+        """
+        Returns True if the taxonomy allows free text tags.
+        """
+        return taxonomy and taxonomy.allow_free_text
+
+    def _check_object(self):
+        """
+        Returns True if this ObjectTag has a valid object.
+
+        Subclasses should override this method to perform any additional validation for the particular type of object tag.
+        """
+        # Must have a valid object id/type:
+        return self.object_id and self.object_type
+
+    def is_valid(self, check_taxonomy=True, check_tag=True, check_object=True) -> bool:
+        """
+        Returns True if this ObjectTag is valid for use with a free-text taxonomy.
+
+        Subclasses should override this method to perform any additional validation for the particular type of object tag.
+
+        If `check_taxonomy` is False, then we skip validating the object tag's taxonomy reference.
+        If `check_tag` is False, then we skip validating the object tag's tag reference.
+        If `check_object` is False, then we skip validating the object ID/type.
+        """
+        # Must be linked to a free-text taxonomy
+        if check_taxonomy and (
+            not self.taxonomy_id or not self.taxonomy.allow_free_text
+        ):
+            return False
+
+        # Open taxonomies don't need an associated tag, but we need a value.
+        if check_tag and not self.value:
```

Comment:
@pomegranited Writing my code I realized that we can also use a `_check_tag` function here, similar to `_check_object`. This is because in the system defined it is only necessary to overwrite the validation of the tag. I think it would be better to add it in your PR

Reply:
Good point -- please see [3e53687](https://github.com/openedx/openedx-learning/pull/62/commits/3e536874b07f5f4cb79b7156f87866b2dafbe10e).

ObjectTag now has 3 methods for this: `_is_valid_taxonomy`, `_is_valid_tag`, and `_is_valid_object`, which subclasses can override. And with the above change, they'll be used whenever `object_tag.is_valid()` is called as well as by `ObjectTag.valid_for`, which the object tag class registry uses to locate a valid subclass.